pkg/actuators/machine/utils.go

```diff
@@ -153,28 +156,42 @@ func correctExistingTags(machine *machinev1.Machine, instance *ec2.Instance, cli
 			if *tag.Key == "kubernetes.io/cluster/"+clusterID && *tag.Value == "owned" {
 				clusterTagOk = true
 			}
+			if _, present := tags[*tag.Key]; present {
+				delete(tags, *tag.Key)
+			}
 		}
 	}
 
-	// Update our tags if they're not set or correct
+	tagsToAdd := []*ec2.Tag{}
+	for key, value := range tags {
+		tagsToAdd = append(tagsToAdd, &ec2.Tag{
+			Key:   aws.String(key),
+			Value: aws.String(value),
+		})
+		klog.Infof("key: %s  value: %s preparing tag to add", key, value)
+	}
+
 	if !nameTagOk || !clusterTagOk {
+		tagsToAdd = append(tagsToAdd, &ec2.Tag{
+			Key:   aws.String("kubernetes.io/cluster/" + clusterID),
+			Value: aws.String("owned"),
+		})
+		tagsToAdd = append(tagsToAdd, &ec2.Tag{
+			Key:   aws.String("Name"),
+			Value: aws.String(machine.Name),
+		})
+	}
+
+	klog.Infof("let of the tags to be updated: %d, %+v", len(tagsToAdd), tagsToAdd)
+	if len(tagsToAdd) != 0 {
 		// Create tags only adds/replaces what is present, does not affect other tags.
 		input := &ec2.CreateTagsInput{
 			Resources: []*string{
 				aws.String(*instance.InstanceId),
 			},
-			Tags: []*ec2.Tag{
-				{
-					Key:   aws.String("kubernetes.io/cluster/" + clusterID),
-					Value: aws.String("owned"),
-				},
-				{
-					Key:   aws.String("Name"),
-					Value: aws.String(machine.Name),
-				},
-			},
+			Tags: tagsToAdd,
 		}
-		klog.Infof("Invalid or missing instance tags for machine: %v; instanceID: %v, updating", machine.Name, *instance.InstanceId)
+		klog.Infof("Missing instance tags for machine: %v; instanceID: %v, updating", machine.Name, *instance.InstanceId)
```

Comment:
This log line is no longer meaningful. Because when the user defined tags are update this line saying instance tags are missing will be printed. Instead print a more meaningful message.

Reply:
thanks Arjun, Now changed the log saying that update the tags and provided the tags which are required to update